### PR TITLE
Starting to add some tests for internal reader

### PR DESCRIPTION
**Pull request by Kirill888**
probably a bunch of things in `odc.stac.testing.{_xr,_cog}` should be generalized and moved to `odc-geo` and made available when `rasterio` and `xarray` are installed.

`_cog.py` is basically copy from `datacube` with dask parts removed, new things are `write_cog_layers` which allows arbitrary overviews to be written, this will be useful when testing read from overviews properly, so far that was only tested in the notebook.

also includes a fix for #46

### CODE REVIEW

File: odc/stac/_eo3converter.py [hunk 1]
```diff
@@ -13,7 +13,12 @@
 import pystac.errors
 import pystac.item
 from datacube.index.eo3 import prep_eo3
-from datacube.index.index import default_metadata_type_docs
+
+try:
+    from datacube.index.index import default_metadata_type_docs  # type: ignore
```
Comment (gadomski) [inline odc/stac/_eo3converter.py:18]: What are the conditions under which `index.default_metadata_type_docs` does not import?

File: tests/test-env-py38.yml [hunk 2]
```diff
@@ -51,4 +54,4 @@ dependencies:
   - pip:
       # dev
       - shed
-      - odc-geo >=0.1.1a2
+      - git+https://github.com/opendatacube/odc-geo
```
Comment (gadomski) [inline tests/test-env-py38.yml]: Should probably have a version and/or commit pin?

**Review by gadomski — COMMENTED** [review]
> probably a bunch of things in odc.stac.testing.{_xr,_cog} should be generalized and moved to odc-geo and made available when rasterio and xarray are installed.

Yeah, I agree with this for sure. This is a lot of custom logic for a `testing` module.

File: tests/test-env-py38.yml (hunk 2 above)
Comment (Kirill888) [reply tests/test-env-py38.yml]: yeah, it's just for now, once new version is released with current fixes and maybe with cog/xr tools I'll switch to that

File: odc/stac/_eo3converter.py (hunk 1 above)
Comment (Kirill888) [reply odc/stac/_eo3converter.py:18]: when running against `develop` branch of datacube.

**Comment by Kirill888** [comment]
@gadomski I have moved those utilities to odc-geo, see https://github.com/opendatacube/odc-geo/pull/32

**Review by gadomski — APPROVED** [review]

**Pull request merged**